transformer_engine/paddle/layer/attention.py

```diff
@@ -567,6 +575,22 @@ def _te_forward(
         set_zero: bool = True,
     ) -> paddle.Tensor:
 
+        if self.deterministic:
+            # workspace optimization path is deterministic
+            os.environ["CUDNN_FRONTEND_ATTN_DP_WORKSPACE_LIMIT"] = "-1"
```

Comment:
Should it be `os.environ["CUDNN_FRONTEND_ATTN_DP_WORKSPACE_LIMIT"] = "1"` ?

Reply:
`-1` means `enable workspace optimization always`. This part is copied directly from pytorch.